docs/_posts/2021-08-31-luau-recap-august-2021.md

```diff
@@ -0,0 +1,112 @@
+* Updating reflection information to provide more specific `Instance` types and correct previously missing or wrong type annotations
+* Hybrid typechecking mode which tries to infer types even in scripts with no typechecking enabled
+* Support for types that are attached to the `DataModel` tree elements to provide instance member information
+* Placing limits to finish typechecking in a finite space/time
+* Adding Autocomplete API for the Roblox Studio to get location-based entity information and appropriate suggestions
+* Additional type inference engine improvements and fixes
+
+While our work continues to respond to the feedback we receive, our team members are shifting focus to add generic functions, improve type refinements in conditionals, extend Parallel Luau, improve Lua VM performance and provide documentation. 
+
+## Typechecking improvements
+
+Type constraint resolver now remembers constraints placed on individual table fields.
+
+This should fix false-positive errors reported after making sure the optional table field is present:
+```lua
+--!strict
+local t: {value: number?} = {value = 2}
+
+if t.value then
+    local v: number = t.value -- ok
+end
+```
+
+And it can also refine field type to a more specific one:
+```lua
+--!strict
+local t: {value: string|number} = {value = 2}
+
+if type(t.value) == "number" then
+    return t.value * 2 -- ok
+end
+```
+
+Like before, combining multiple conditions using 'and' and 'not' is also supported.
+
+---
+
+Constructing arrays with different values for optional/union types are now also supported for individual table fields and in functions call arguments:
+```lua
+--!strict
+type Foo = {x: number | string, b: number?}
+
+local function foo(l: {Foo}) end
+
+foo({
+	{x = 1234567},
+	{x = "hello"}, -- now ok
+})
+
+type Bar = {a: {Foo}}
+
+local foos: Bar = {a = {
+	{x = 1234567},
+	{x = "hello", b = 2}, -- now ok
+}}
+```
+
+---
+
+Finally, we have fixed an issue with Roblox class field access using indexing like `part["Anchored"] = true`.
+
+## Linter improvements
+
+We have added a new linter check for duplicate local variable definitions.
+
+It is created to find duplicate names in cases like these:
+```lua
+local function foo(a1, a2, a2) -- Function argument 'a2' already defined on column 24
+local a1, a2, a2 = f() -- Variable 'a2' already defined on column 11
+
+local bar = {}
+function bar:test(self) -- Function argument 'self' already defined implicitly
+```
+
+Our UnknownType linter warning was extended to check for correct class names passed into `FindFirstChildOfClass`, `FindFirstChildWhichIsA`, `FindFirstAncestorOfClass` and `FindFirstAncestorWhichIsA` funcitons.
+
+## Performance improvements
+
+We have added an optimization to 'table.unpack' for 2x performance improvement.
+
+One extra optimization for tables is to predict required table size based on fields that are assigned to it in the code after construction.
```

Comment:
```suggestion
One extra optimization for tables is to predict required table capacity based on fields that are assigned to it in the code after construction. This can reduce the need to reallocate tables.
```